frontend/src/views/upload-project/UploadProjectView.tsx

```diff
@@ -50,37 +68,58 @@ export const UploadProjectView = () => {
         file: null,
       },
     };
+    let response: any;
     try {
-      const response = await graphQLHttpClient.sendFile(uploadProjectMutation, variables, file);
-      const action = { type: HANDLE_SUBMIT__ACTION, response };
-      dispatch(action);
+      dispatch({ type: 'HANDLE_UPLOAD' });
+      response = await graphQLHttpClient.sendFile(uploadProjectMutation, variables, file);
+      if (response?.data?.uploadProject?.message) {
+        dispatch({ type: 'SHOW_TOAST', message: response.data.uploadProject.message });
+      } else {
+        dispatch({ type: 'SHOW_TOAST', message: 'An unexpected error has occurred, please refresh the page' });
+      }
     } catch (exception) {
-      const action = { type: HANDLE_SUBMIT__ACTION, response: { error: exception.toString() } };
-      dispatch(action);
+      dispatch({ type: 'SHOW_TOAST', message: exception.toString() });
     }
+    dispatch({ type: 'HANDLE_RESPONSE', response });
   };
 
-  const onFileSelected = (file) => {
-    const action = { type: HANDLE_SELECTED__ACTION, file };
-    dispatch(action);
+  const onFileSelected = (file: string) => {
```

Comment:
Are you really sure that the type is string?

Reply:
It is a File, see the [W3C specification](https://w3c.github.io/FileAPI/#file-section)